src/web/app/components/question-submission-form/question-submission-form.component.ts

```diff
@@ -98,6 +98,27 @@ export class QuestionSubmissionFormComponent implements OnInit {
   }
 
   ngOnInit(): void {
+    this.sortRecipientsByName();
+  }
+
+  /**
+   * Sorts recipients of feedback by their name.
+   */
+  private sortRecipientsByName(): void {
+    this.model.recipientList.sort((firstRecipient: FeedbackResponseRecipient,
+      secondRecipient: FeedbackResponseRecipient) =>
+      firstRecipient.recipientName.localeCompare(secondRecipient.recipientName));
+ 
+    const indexes: Map<String, number> = new Map();
+    this.model.recipientList.forEach((recipient: FeedbackResponseRecipient, index: number) => {
+      indexes.set(recipient.recipientIdentifier, index);
+    });
+
+    this.model.recipientSubmissionForms.sort((firstRecipient: FeedbackResponseRecipientSubmissionFormModel,
+      secondRecipient: FeedbackResponseRecipientSubmissionFormModel) => {
+
+      return indexes.get(firstRecipient.recipientIdentifier)! - indexes.get(secondRecipient.recipientIdentifier)!;
```

Comment:
@daongochieu281 Any advice on how to get `Object is possibly 'undefined'`, realised that non-null assertion operators are not allowed. I think the assumption in the sort is that the recipientList always contains all the emails. I could return a dummy value since this case should never happen, but am worried that there would be unintended consequences in the future

Reply:
you can try `const name = indexes.get(firstRecipient.recipientIdentifier) || ''`

Reply:
Since its an index, 

const index = indexes.get(firstRecipient.recipientIdentifier) || 0

Would it be ok to set the index to 0 as the default value. Only problem is if the list of emails incorrectly omits the identifier, this would cause the sort to fail.

Reply:
`recipientIdentifier` should not be null, so I think its ok